static/js/apps/eval_retrieval_generation/sxs/app.tsx

```diff
@@ -41,14 +42,38 @@ function getSortedQueryIds(docInfos: { a: DocInfo; b: DocInfo }) {
     .sort((a, b) => a - b);
 }
 
+/** Returns the ID of the first unevaluated query. */
+async function getStartingQueryId(
+  props: AppPropType,
+  sortedQueryIds: number[]
+): Promise<number> {
+  const completedIds = await getRatedQueryIds(
+    props.sheetIdA,
+    props.sheetIdB,
+    props.sessionId
+  );
+  for (const queryId of sortedQueryIds) {
+    if (!completedIds.includes(queryId)) {
+      return queryId;
+    }
+  }
+  // If all evals are complete, show the first query.
+  return sortedQueryIds[0];
+}
+
 export function App(props: AppPropType): JSX.Element {
   const [user, setUser] = useState<User | null>(null);
   const [docInfos, setDocInfos] = useState<{ a: DocInfo; b: DocInfo }>(null);
   const { setSessionQueryId, sessionQueryId } = useContext(SessionContext);
+  console.log(sessionQueryId);
+
   const sortedQueryIds = getSortedQueryIds(docInfos);
   if (!sessionQueryId && sortedQueryIds.length) {
-    setSessionQueryId(sortedQueryIds[0]);
+    getStartingQueryId(props, sortedQueryIds).then(
```

Comment:
nit: is it safe to assume we should be doing the logic to get startingQueryId anytime docInfos changes? which in that case, instead of getting sortedQueryIds and doing this check everytime this component re-renders (whenever state or context changes), we should use a useEffect instead like: 

```
useEffect(() => {
  const sortedQueryIds = ...
  if (sortedQueryIds.length) {
     ...
  }
}, [docInfos])
```

Reply:
Updated to a useEffect and also changed to calculate sortedQueryIds only once after docInfos are fetched.